### DISCUSSION TITLE
Which desktop UI framework works with AOT ?

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/runtime#93509] [Q&A]
WPF/Winforms don't support it. Is there any other UI framework for desktop apps which officially supports AOT ?

**Comment by gfoidl** [comment]
E.g. [Avalonia UI](https://docs.avaloniaui.net/).

**Reply by Symbai** [reply]
I tested it and it works but it creates 20mb large files which is the reason I'm looking for something else but thanks.

**Comment by KeterSCP** [comment]
WinForms app can work with AOT
See https://github.com/kant2002/WinFormsComInterop

**Reply by Symbai** [reply]
Can.. but also cannot. Its not official supported which means it can be broken at any time. Speaking about that I read an issue about it stopped working with .NET 8.

**Reply by KeterSCP** [reply]
It is worth trying first. I have a .NET8 winforms app which works perfectly with that package in AOT

**Comment by NCLnclNCL** [comment]
.
